lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -888,12 +888,14 @@ defmodule Kernel.ErrorsTest do
   end
 
   test "def defmacro clause change" do
-    assert_eval_raise CompileError, "nofile:3: defmacro foo/1 already defined as def", '''
-    defmodule Kernel.ErrorsTest.DefDefmacroClauseChange do
-      def foo(1), do: 1
-      defmacro foo(x), do: x
-    end
-    '''
+    assert_eval_raise CompileError,
+                      "nofile:3: defmacro foo/1 already defined as def in nofile:3",
```

Comment:
You are using the wrong File:Line, given it is just being repeated in the error message. You would have to pick the other File:Line pairing.

Reply:
You are right.  Fixed!